File: app/src/test/java/info/nightscout/androidaps/utils/PercentageSplitterTest.java
package info.nightscout.androidaps.utils;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by mike on 22.12.2017.
 */

public class PercentageSplitterTest {
    public PercentageSplitterTest() {
        super();
    }

    @Test
    public void pureNameTestPercentageOnly() {
        assertEquals("Fiasp", PercentageSplitter.pureName("Fiasp(101%)"));
    }

    @Test
    public void pureNameTestPercentageAndPositiveTimeShift() {
        assertEquals("Fiasp", PercentageSplitter.pureName("Fiasp (101%,2h)"));
    }

    @Test
    public void pureNameTestPercentageAndNegtiveTimeShift() {
        assertEquals("Fiasp", PercentageSplitter.pureName("Fiasp (50%,-2h)"));
    }
}

File: app/src/test/java/info/nightscout/androidaps/utils/PercentageSplitterTest.kt
package info.nightscout.androidaps.utils

import info.nightscout.androidaps.utils.PercentageSplitter.pureName
import org.junit.Assert
import org.junit.Test

/**
 * Created by mike on 22.12.2017.
 */
class PercentageSplitterTest {

    @Test fun pureNameTestPercentageOnly() {
        Assert.assertEquals("Fiasp", pureName("Fiasp(101%)"))
    }

    @Test fun pureNameTestPercentageAndPositiveTimeShift() {
        Assert.assertEquals("Fiasp", pureName("Fiasp (101%,2h)"))
    }

    @Test fun pureNameTestPercentageAndNegtiveTimeShift() {
        Assert.assertEquals("Fiasp", pureName("Fiasp (50%,-2h)"))
    }
}